lib/vanagon/project.rb

```diff
@@ -270,14 +270,24 @@ def has_provides?
       !get_provides.empty?
     end
 
-    # Collects the preinstall packaging actions for the project and it's components
-    # for the specified packaging state
+    # Checks that the pkg_state is valid (install OR upgrade), return vanagon error if invalid
+    #
+    # @param pkg_state [String] package state input
+    def check_pkg_state(pkg_state)
+      unless ["install", "upgrade"].include? pkg_state
+        raise Vanagon::Error, "#{pkg_state} should be a string containing one of 'install' or 'upgrade'"
+      end
+    end
```

Comment:
Looks like there are two separate implementations of this. Wondering if there's somewhere good we can add this more generically so we can define it once and use it in all the places! Maybe in `Vanagon::Utilities`?

Reply:
One checks for an array containing 1 or more of install/upgrade, the other checks that the string is either install or upgrade. Would we still want to combine it?

Reply:
ah, woops. Didn't read the code closely enough. I'm trying to think of a neat way to get these combined still but nothing is coming to mind, so could the methods have different names? I'm not sure the best way to name it unambiguously, but methods for two different things with the same name is a little confusing :)

Reply:
Yeah for sure, I'll make that change!

Reply:
Would "check_project_pkg_state" and "check_dsl_pkg_state" be alright?

Reply:
Hm, no, that still seems confusing.

Maybe having `_string` vs `_array` in the names? I think mostly this is confusing for me because we're only checking for a subset of the possible package states (usually things will either happen on (install, upgrade, or install and upgrade) or (removal, upgrade, or removal and upgrade) I don't have any great suggestions for how to make this better though, might need to think on it more.

Reply:
Ok cool cool, I'll change those names and if anything I can always make a ticket to combine those later when we've had more time to think about it